apps/epic-web/src/inngest/functions/sanity/product/sanity-product-created.ts

```diff
@@ -82,6 +83,114 @@ export const sanityProductCreated = inngest.createFunction(
       })
     })
 
+    type SplitInfo = {
+      type: string
+      percent: number
+      userId: string | undefined | null
+    }
+
+    type Splits = {
+      skill: SplitInfo
+      owner: SplitInfo
+      contributor?: SplitInfo
+    }
+
+    const defineSplits = await step.run(
+      'define splits',
+      async (): Promise<Splits> => {
+        const OWNER_ID = '4ef27e5f-00b4-4aa3-b3c4-4a58ae76f50b'
+        const isOwnerInstructor = instructor?.userId === OWNER_ID
+        const isSelfPaced = type === 'self-paced'
+
+        let splits: Splits = {
+          skill: {
+            type: 'skill',
+            percent: 0,
+            userId: null,
+          },
+          owner: {
+            type: 'owner',
+            percent: 0,
+            userId: OWNER_ID,
+          },
+        }
+
+        if (isOwnerInstructor) {
+          if (isSelfPaced) {
+            splits.skill.percent = 0.4
+            splits.owner.percent = 0.6
+          } else {
+            splits.skill.percent = 0.15
+            splits.owner.percent = 0.85
+          }
+        } else {
+          if (isSelfPaced) {
+            splits.skill.percent = 0.5
+            splits.owner.percent = 0.2
+            splits.contributor = {
+              type: 'contributor',
+              percent: 0.3,
+              userId: instructor?.userId,
+            }
+          } else {
+            splits.skill.percent = 0.15
+            splits.owner.percent = 0.15
+            splits.contributor = {
+              type: 'contributor',
+              percent: 0.7,
+              userId: instructor?.userId,
+            }
+          }
+        }
```

Comment:
This is kind of harsh, but I am concerned with it's hard-coding very specific business values and mutating an object through another object. To the extent that I think we might need to rethink the entire data model and implementation of the PR.

We can keep trying like this or write a plan, diagram the flows, etc before implementing

I'd rather have it just manually added to the database. 😬